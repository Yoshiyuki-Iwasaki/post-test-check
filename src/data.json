[
    {
        "id": 1,
        "username": "username01",
        "date": "date01",
        "content": "投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容"
    },
    {
        "id": 2,
        "username": "username02",
        "date": "date02",
        "content": "投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容"
    },
    {
        "id": 3,
        "username": "username03",
        "date": "date03",
        "content": "投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容"
    }
]