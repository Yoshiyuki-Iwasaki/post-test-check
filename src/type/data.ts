

export type postDataType = {
  id: number;
  username: string;
  date: string;
  content: string;
};
