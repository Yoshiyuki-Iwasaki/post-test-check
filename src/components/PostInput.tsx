import React from 'react'
import styled from "styled-components";
import { postDataType } from "../type/data";
import useInput from "../hooks/useInput";

const Form = styled.form`
  margin: 0 auto;
  width: 400px;
`;
const Input = styled.input`
  width: 100%;
  height: 80px;
`;

type PostInputType = {
  postData: postDataType[];
  setPostData: any;
};

const PostInput: React.FC<PostInputType> = ({ postData, setPostData }) => {
  const attrs = useInput("");

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (!attrs.value) return;
    setPostData([
      ...postData,
      {
        id: postData.length + 1,
        username: `username${postData.length + 1}`,
        date: "date",
        content: attrs.value,
      },
    ]);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <Input type="text" {...attrs} />
    </Form>
  );
};

export default PostInput
