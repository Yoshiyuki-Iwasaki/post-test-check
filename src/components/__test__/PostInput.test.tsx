import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import PostInput from "../PostInput";
import data from "../../postData.json";

describe("<PostInput>", () => {
  const props = {
    postData: data,
    setPostData: jest.fn(),
  };
  test("Snapshotは問題ないか", () => {
    const { asFragment } = render(<PostInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
