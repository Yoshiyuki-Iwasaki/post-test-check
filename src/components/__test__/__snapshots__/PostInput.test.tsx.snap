// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostInput> Snapshotは問題ないか 1`] = `
<DocumentFragment>
  <form
    class="sc-bdvvtL jApnZM"
  >
    <input
      class="sc-gsDKAQ cnUKyX"
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
